### PR TITLE
Add post_filter is supported in hybrid search

### PR DESCRIPTION
**Pull request by vibrantvarun**
### Description
Post Filter usage in hybrid search.

### Issues Resolved
Closes [6646](https://github.com/opensearch-project/documentation-website/issues/6646)

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kolchfa-aws** [comment]
@vibrantvarun I have reviewed and pushed my edits to the same PR. If you'd like to look over my edits to see that everything looks good, I'll move the PR to editorial review. Once the editor reviews, I'll implement his comments and will merge the PR.

**Review by vibrantvarun — COMMENTED** [review]
Looks good to me

File: _search-plugins/hybrid-search.md [hunk 1]
```diff
@@ -216,3 +218,159 @@ The response contains the matching document:
...
+You can perform post-filtering on hybrid search results by providing the `post_filter` parameter in your query.
+
+The `post_filter` clause is applied after the search results have been retrieved. Post-filtering is useful for applying additional filters to the search results without impacting the scoring or the order of the results. 
+
+Post-filtering does not impact document relevance scores or aggregation results.
+{: .note}
+
+#### Example: Post-filtering
+
+The following example request combines two query clauses---a `term` query and a `match` query. This is the same query as the [preceding example](#example-combining-a-match-query-and-a-term-query) but it contains a `post_filter`:
```
Comment (natebower) [inline _search-plugins/hybrid-search.md:313]:
```suggestion
The following example request combines two query clauses---a `term` query and a `match` query. This is the same query as in the [preceding example](#example-combining-a-match-query-and-a-term-query), but it contains a `post_filter`:
```

**Review by natebower — COMMENTED** [review]
@vibrantvarun @kolchfa-aws Just a couple minor changes. Thanks!

**Review by kolchfa-aws — APPROVED** [review]
LGTM. Thank you, @vibrantvarun!

**Pull request merged**